src/StructuredLogger/ObjectModel/Build.cs

```diff
@@ -288,5 +289,18 @@ public void WaitForBackgroundTasks()
                 backgroundTasks.Clear();
             }
         }
+
+        /// <summary>
+        /// Indicates whether current log/build is of a higher than known version
+        ///  or that there were unknown events or data detected during reading.
+        /// </summary>
+        public bool IsCompatibilityMode { get; internal set; }
+
+        /// <summary>
+        /// In case forward compatible reading was allowed and encountered recoverable errors, this method
+        ///  will return the list of errors encountered during reading.
+        /// </summary>
+        public IReadOnlyList<(ReaderErrorType errorType, BinaryLogRecordKind recordKind, int count)>

```

Comment:
Here I'd actually make a class instead of a ValueTuple

Reply:
There are performance and semantic differences.

A class is a reference type a `ValueTuple` is a value type. It's the difference between one "big" heap allocation for the backing array and one for each element or just a "big" heap allocation for everything.

C# tuples are meant to be bags of related values that do not represent an entity. They are usually deconstructed on "read".

Reply:
Removing the code alltogether